rebasehelper/helpers/lookaside_cache_helper.py

```diff
@@ -112,14 +112,14 @@ def _download_source(cls, tool, url, package, filename, hashtype, hsh, target=No
             raise LookasideCacheError(str(e))
 
     @classmethod
-    def download(cls, tool, basepath, package):
+    def download(cls, tool, basepath, package, target=None):
```

Comment:
If you are adding support for downloading sources elsewhere, you should also add support for uploading them from elsewhere.

Reply:
That is already supported, isn't it? `LookasideCacheHelper._upload_source()` has an argument `filename` which can be both relative and absolute.

Reply:
No, there are never absolute paths in `sources` file, and there shouldn't be.

Reply:
Maybe I should rather say, there are no paths in `sources`, only filenames.